flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/PipelinedRegionSchedulingStrategy.java

```diff
@@ -127,13 +136,9 @@ private void maybeScheduleRegion(final SchedulingPipelinedRegion region) {
 
 		checkState(areRegionVerticesAllInCreatedState(region), "BUG: trying to schedule a region which is not in CREATED state");
 
-		final Set<ExecutionVertexID> verticesToSchedule = IterableUtils.toStream(region.getVertices())
-			.map(SchedulingExecutionVertex::getId)
-			.collect(Collectors.toSet());
 		final List<ExecutionVertexDeploymentOption> vertexDeploymentOptions =
-			SchedulingStrategyUtils.createExecutionVertexDeploymentOptionsInTopologicalOrder(
-				schedulingTopology,
-				verticesToSchedule,
+			SchedulingStrategyUtils.createExecutionVertexDeploymentOptions(
```

Comment:
should we also do this optimisation for other strategies to compare performance in a cleaner way?

Reply:
I think it's not needed. Because the sorting is not a problem for other strategies where the sorting is invoked once per triggering event. While for pipelined region scheduling, it is invoked multiple times, each for one region.
And doing it like this requires to add a sorted region vertex map for other strategies.